ts/client/features/genesis.ts

```diff
@@ -17,14 +22,49 @@ export class GenesisCommitment extends BaseCommitment {
     }
 }
 
+const genesisBatchID = 0;
+
 export class GenesisHandlingStrategy implements BatchHandlingStrategy {
-    constructor(private genesisStateRoot: BytesLike) {}
-    async parseBatch(event: Event): Promise<Batch> {
+    constructor(
+        private readonly rollup: Rollup,
+        private genesisStateRoot: BytesLike
+    ) {}
+
+    private validateIsFirstBatch(batchID: number) {
+        if (batchID !== genesisBatchID) {
+            throw new NotFirstBatch();
+        }
+    }
+
+    public async parseBatch(event: Event): Promise<Batch> {
+        const batchIDBN = event.args?.batchID as BigNumber;
+        const batchID = batchIDBN.toNumber();
+
+        this.validateIsFirstBatch(batchID);
+
+        // Create L2 state
```

Comment:
Nit: What do you think we rephrase this as something like "reconstruct genesis batch from genesis state root"?